src/components/common/AddDiscountCodeForm.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <div class="row">
+    <div class="col-sm-offset-9 col-sm-3">
+      <ServerError :error="serverError">
+        <template slot-scope="{ graphQLError }">
+          {{ getErrorMessage(graphQLError) }}
+        </template>
+      </ServerError>
+    </div>
+    <div class="col-sm-12">
+      <form @submit.prevent="submitDiscountCode"
+            class="text-right promotion-info">
+        <span class="text-uppercase promo-info-text">
+          {{ $t('code') }}
+          <img src="../../assets/img/information.png"
+               class="info-icon"
+               alt="information icon">
+        </span>
+        <ValidationError :vuelidate="$v.code"
+                         class="vuelidate">
+          <input v-model.trim.lazy="$v.code.$model"
+                 id="promo-code"
+                 type="text"
+                 data-test="discount-code-input"/>
+        </ValidationError>
+        <LoadingButton :buttonState="buttonState"
+                       class="apply-button"
+                       data-test="apply-discount-code-button">
+          {{ $t('apply') }}
+        </LoadingButton>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+import gql from 'graphql-tag';
+import { required } from 'vuelidate/lib/validators';
+import LoadingButton from './LoadingButton.vue';
+import ServerError from './ServerError.vue';
+import cartMixin from '@/mixins/cartMixin';
+import ValidationError from './ValidationError.vue';
+
+export default {
+  components: { LoadingButton, ServerError, ValidationError },
+
+  data: () => ({
+    me: null,
+    code: null,
+    buttonState: null,
+    serverError: null,
+  }),
+
+  methods: {
+    addDiscountCode() {
+      return this.updateMyCart({
+        addDiscountCode: {
+          code: this.code,
+        },
+      });
+    },
+
+    submitDiscountCode() {
+      this.serverError = null;
+      if (!this.$v.$invalid) {
+        this.buttonState = 'loading';
+        this.addDiscountCode()
+          .then(() => {
+            this.buttonState = 'success';
+          })
+          .catch((error) => {
+            this.serverError = error;
+            this.buttonState = null;
+          });
+      }
+    },
+
+    getErrorMessage({ code }) {
+      if (code === 'DiscountCodeNonApplicable') {
+        return this.$t('nonApplicable');
+      }
+      return this.$t('unknownError');
+    },
+  },
+
+  mixins: [cartMixin],
+
+  apollo: {
+    me: {
+      query: gql`
+        query me($locale: Locale!){
+          me {
+            activeCart {
+              id
+              version
+              discountCodes {
+                discountCode {
+                  id
+                  code
+                  name(locale: $locale)
+                }
+              }
```

Comment:
I guess I skipped this part before, but I cannot find where you are using this info in this component. After making the PR, review every query to make sure you are not over-fetching, so that the performance is not affected :)